typescript/src/Common.ts

```diff
@@ -67,6 +67,24 @@ function prepareIt(): void {
             }
         });
     }
+
+    $("input[required]").bind("invalid", (e) => {
+        const el = e.currentTarget;
```

Comment:
This handler function should be extracted into a named and exported function. Because the `invalid` event does not bubble, the handler for the event must be attached to each individual element where the event should get custom handling. This code will attach the handler to only elements that exist at the time the DOM `ready` event fires, so dynamically added elements with a `required` attribute must be able to add this handler at the time the element is injected to get the same functionality.